### PR TITLE
feat(OX-3565) add oxeye scan action

### PR DESCRIPTION
**Pull request by oxeye-schmil**

### CODE REVIEW

File: images/Dockerfile.oxeye-scan [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# should the same as semgrep_config_base_dir in entrypoint-sast.sh
+ARG SEMGREP_CONFIG_DIR="/"
+
+################################################
+#                 Build OXCTL                  #
+################################################
+FROM golang:1.19-alpine3.18 AS oxctl_builder
```
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:7]: Let's use [ecr images ](https://gallery.ecr.aws/docker/library/golang) to save dockerhub quotas

File: images/Dockerfile.oxeye-scan [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+RUN git config --global --add url."https://${GH_PERSONAL_ACCESS_TOKEN}:x-oauth-basic@github.com/".insteadOf "https://github.com/"
+RUN git clone --depth 1 --branch v0.0.2 https://github.com/ox-eye/oxctl.git /app
+WORKDIR /app
+RUN go build
+
+################################################
+#        download semgrep rules & syft         #
+################################################
+FROM alpine:3.18.0 AS downloader
```
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:21]: Let's use [ecr images ](https://gallery.ecr.aws/docker/library/alpine) to save dockerhub quotas

File: images/Dockerfile.oxeye-scan [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+# downlaod semgrep rules
+ARG SEMGREP_CONFIG_DIR
+ARG SEMGREP_RULES_COMMIT="52874aa7acd74af3d966a26b170aa08714578df5"
+
+RUN mkdir $SEMGREP_CONFIG_DIR && \
+    apk add git && \
```
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:28]: Let's add --no-cache to keep cache smaller

File: images/Dockerfile.oxeye-scan [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    git clone https://github.com/returntocorp/semgrep-rules $SEMGREP_CONFIG_DIR/semgrep-rules && \
```
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:29]: We can do cloning here directly on specific SHA\REF (see 13th line)

File: images/Dockerfile.oxeye-scan [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+RUN wget -O- https://raw.githubusercontent.com/anchore/syft/${SYFT_VERSION}/install.sh | sh -s -- -b /usr/local/bin ${SYFT_VERSION}
+
+################################################
+#              Build final image               #
+################################################
+FROM alpine:3.18
+
+# install semgrep
+ARG SEMGREP_CONFIG_DIR
+RUN apk update && apk add --update --no-cache \
```
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:45]: Let's add `rm -rf /var/cache/apk/*` to remove updated packages' indices to keep final image smaller

File: images/Dockerfile.oxeye-scan [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+COPY semgrepignore $SEMGREP_CONFIG_DIR/.semgrepignore
+
+# downlaod syft
+ARG SYFT_VERSION="v0.70.0"
+RUN wget -O- https://raw.githubusercontent.com/anchore/syft/${SYFT_VERSION}/install.sh | sh -s -- -b /usr/local/bin ${SYFT_VERSION}
+
+################################################
+#              Build final image               #
+################################################
+FROM alpine:3.18
```
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:41]: Let's use [ecr images ](https://gallery.ecr.aws/docker/library/alpine)to save dockerhub quotas

File: images/Dockerfile.oxeye-scan [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    gcc \
+    jq \
+    musl-dev \
+    python3-dev \
+	python3 \
+	py3-pip
+
+COPY --from=oxctl_builder /app/oxctl /
+COPY --from=downloader $SEMGREP_CONFIG_DIR $SEMGREP_CONFIG_DIR
+RUN pip install semgrep==1.22.0
```
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:56]: Let's use `--no-cache-dir` to keep final image smaller

**Review by DovnarAlexander — CHANGES_REQUESTED** [review]

File: images/Dockerfile.oxeye-scan (hunk 6 above)
Comment (DovnarAlexander) [inline images/Dockerfile.oxeye-scan:41]: Let's use [ecr images ](https://gallery.ecr.aws/docker/library/alpine)to save dockerhub quotas

**Review by DovnarAlexander — CHANGES_REQUESTED** [review]

File: images/Dockerfile.oxeye-scan (hunk 2 above)
Comment (oxeye-schmil) [reply images/Dockerfile.oxeye-scan:21]: done

File: images/Dockerfile.oxeye-scan (hunk 1 above)
Comment (oxeye-schmil) [reply images/Dockerfile.oxeye-scan:7]: done

File: images/Dockerfile.oxeye-scan (hunk 3 above)
Comment (oxeye-schmil) [reply images/Dockerfile.oxeye-scan:28]: done

File: images/Dockerfile.oxeye-scan (hunk 5 above)
Comment (oxeye-schmil) [reply images/Dockerfile.oxeye-scan:45]: done

File: images/Dockerfile.oxeye-scan (hunk 4 above)
Comment (oxeye-schmil) [reply images/Dockerfile.oxeye-scan:29]: We want to get the most updated rules every time

**Review by DovnarAlexander — APPROVED** [review]

**Pull request merged**